tools/linker/CoreTypeMapStep.cs

```diff
@@ -97,10 +177,18 @@ void SetIsDirectBindingValue (TypeDefinition type)
 			// * https://bugzilla.xamarin.com/show_bug.cgi?id=15465
 			if (IsCIFilter (type)) {
 				isdirectbinding_value [type] = null;
+#if NET
+				var base_type = Context.Resolve (type.BaseType);
+#else
 				var base_type = type.BaseType.Resolve ();
```

Comment:
Creating an extension method on LinkContext for legacy Xamarin would make it possible to avoid all this conditional code:

```csharp
public static TypeDefinition Resolve (this LinkContext context, TypeReference type)
{
    return type.Resolve ();
}

// and now the following code works for both legacy Xamarin and .NET:

var base_type = Context.Resolve (type.BaseType);
```